src/libcore/str/pattern.rs

```diff
@@ -450,21 +450,15 @@ impl<'a> Pattern<'a> for char {
 
     #[inline]
     fn is_prefix_of(self, haystack: &'a str) -> bool {
-        if let Some(ch) = haystack.chars().next() {
-            self == ch
-        } else {
-            false
-        }
+        let mut buffer = [0u8; 4];
+        self.encode_utf8(&mut buffer).is_prefix_of(haystack)
```

Comment:
I think this can just be simplified to `self.encode_utf8(&mut [0; 4]).is_prefix_of(haystack)`? And similarly for below.

Reply:
You are right; I added a cleanup commit.